src/ruuvitag_ble/parser.py

```diff
@@ -23,14 +25,25 @@ def _start_update(self, service_info: BluetoothServiceInfo) -> None:
             return None
 
         data_format = raw_data[0]
-        if data_format != 0x05:
+        if data_format not in (0x03, 0x05):
             _LOGGER.debug("Data format not supported: %s", raw_data)
             return
 
-        decoder = DataFormat5Decoder(raw_data)
+        decoder_classes: dict[
+            int,
+            Type[Union[DataFormat3Decoder, DataFormat5Decoder]],
+        ] = {
+            0x03: DataFormat3Decoder,
+            0x05: DataFormat5Decoder,
+        }
+
+        decoder: Union[DataFormat3Decoder, DataFormat5Decoder] = decoder_classes.get(
+            data_format,
+            DataFormat5Decoder,
+        )(raw_data)
```

Comment:
The "Data format not supported" check from above becomes unnecessary (and more performant) if it was a try-except here.

```suggestion
        try:
            decoder_cls: Union[DataFormat3Decoder, DataFormat5Decoder] = decoder_classes[data_format]
        except KeyError:
            _LOGGER.debug("Data format not supported: %s", raw_data)
            return
        decoder = decoder_cls(raw_data)
```